Tests/SourceKitLSPTests/SemanticTokensTests.swift

```diff
@@ -0,0 +1,687 @@
+    ])
+  }
+
+  func testEmpty() {
+    let text = ""
+    let tokens = openAndPerformSemanticTokensRequest(text: text)
+    XCTAssertEqual(tokens, [])
+  }
+
+  func testRanged() {
+    let text = """
+    let x = 1
+    let test = 20
+    let abc = 333
+    let y = 4
+    """
+    let start = Position(line: 1, utf16index: 0)
+    let end = Position(line: 2, utf16index: 5)
+    let tokens = openAndPerformSemanticTokensRequest(text: text, range: start..<end)
+    XCTAssertEqual(tokens, [
+      makeToken(line: 1, utf16index: 0, length: 3, kind: .keyword),
+      makeToken(line: 1, utf16index: 4, length: 4, kind: .identifier),
+      makeToken(line: 1, utf16index: 11, length: 2, kind: .number),
+      makeToken(line: 2, utf16index: 0, length: 3, kind: .keyword),
+      makeToken(line: 2, utf16index: 4, length: 3, kind: .identifier),
+    ])
+  }
+
+  func testLexicalTokens() {
+    let text = """
+    let x = 3
+    var y = "test"
+    /* abc */ // 123
+    """
+    let tokens = openAndPerformSemanticTokensRequest(text: text)
+    XCTAssertEqual(tokens, [
+      // let x = 3
+      makeToken(line: 0, utf16index: 0, length: 3, kind: .keyword),
+      makeToken(line: 0, utf16index: 4, length: 1, kind: .identifier),
+      makeToken(line: 0, utf16index: 8, length: 1, kind: .number),
+      // var y = "test"
+      makeToken(line: 1, utf16index: 0, length: 3, kind: .keyword),
+      makeToken(line: 1, utf16index: 4, length: 1, kind: .identifier),
+      makeToken(line: 1, utf16index: 8, length: 6, kind: .string),
+      // /* abc */ // 123
+      makeToken(line: 2, utf16index: 0, length: 9, kind: .comment),
+      makeToken(line: 2, utf16index: 10, length: 6, kind: .comment),
+    ])
+  }
+
+  func testLexicalTokensForMultiLineComments() {
+    let text = """
+    let x = 3 /*
+    let x = 12
+    */
+    """
+    let tokens = openAndPerformSemanticTokensRequest(text: text)
+    XCTAssertEqual(tokens, [
+      makeToken(line: 0, utf16index: 0, length: 3, kind: .keyword),
+      makeToken(line: 0, utf16index: 4, length: 1, kind: .identifier),
+      makeToken(line: 0, utf16index: 8, length: 1, kind: .number),
+      // Multi-line comments are split into single-line tokens
+      makeToken(line: 0, utf16index: 10, length: 2, kind: .comment),
+      makeToken(line: 1, utf16index: 0, length: 10, kind: .comment),
+      makeToken(line: 2, utf16index: 0, length: 2, kind: .comment),
+    ])
+  }
+
+  func testLexicalTokensForDocComments() {
+    let text = """
+    /** abc */
+      /// def
+    """
+    let tokens = openAndPerformSemanticTokensRequest(text: text)
+    XCTAssertEqual(tokens, [
+      makeToken(line: 0, utf16index: 0, length: 10, kind: .comment, modifiers: [.documentation]),
+      makeToken(line: 1, utf16index: 2, length: 7, kind: .comment, modifiers: [.documentation]),
+    ])
+  }
+
+  func testLexicalTokensForBackticks() {
+    let text = """
+    var `if` = 20
+    let `else` = 3
```

Comment:
Interestingly, this already seems to work correctly as-is, since the parser won't include the backtick in the identifier if it is only present on one side (probably due to it being a syntax error). I will add a test case that verifies this.